<launch>
  <!-- corner visualization -->
  <node name="dvs_renderer_corner_soft" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="5"/>
    <remap from="events" to="/dvs_corner_events_soft" />
    <remap from="image" to="/dvs/image_raw" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_rendering_corners_soft"/>
  </node>

  <!-- event visualization -->
  <node name="dvs_renderer_edge_soft" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/dvs_edge_events_soft" />
    <remap from="image" to="/dvs/image_raw" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_rendering_edges_soft"/>
  </node>


  <!-- event visualization -->
  <node name="dvs_renderer_noise_soft" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/dvs_noise_events_soft" />
    <remap from="image" to="/dvs/image_raw" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_rendering_noises_soft"/>
  </node>


<node name="dvs_renderer_raw" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/dvs/events" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_renderer_raw"/>
  </node>

<node name="dvs_renderer_center" pkg="dvs_renderer" type="dvs_renderer">
    <param name="display_method" value="red-blue"/>
    <param name="num_event_msgs" value="1"/>
    <remap from="events" to="/object_center" />
    <remap from="camera_info" to="/dvs/camera_info" />
    <remap from="dvs_rendering" to="dvs_renderer_raw"/>
  </node>

  <!-- display -->
  <node name="rqt_gui_corners" pkg="rqt_gui" type="rqt_gui" args="--perspective-file $(find visual_servoing_davis)/cfg/soft_exp.perspective" />
  
  <!-- camera driver -->  
  <node name="davis_ros_driver" pkg="davis_ros_driver" type="davis_ros_driver">
  </node>

  <rosparam file="$(find davis_ros_driver)/config/ARIC_VS.yaml" />

  <!-- feature detector -->
  <!-- <node name="visual_servoing_node" pkg="visual_servoing_davis" type="visual_servoing_node" output="screen">
  </node>  -->




</launch>
